frontend/src/components/pages/connect/Overview.tsx

```diff
@@ -27,9 +27,50 @@ import RpConnectPipelinesList from '../rp-connect/Pipelines.List';
 import { RedpandaConnectIntro } from '../rp-connect/RedpandaConnectIntro';
 import { Features } from '../../../state/supportedFeatures';
 import {isServerless} from '../../../config';
+import RpConnectSecretsList from '../rp-connect/secrets/Secrets.List';
+import { useLocation } from 'react-router-dom';
+
+enum ConnectView {
+    kafkaConnect = 'kafka-connect',
+    RedpandaConnect = 'redpanda-connect',
+    RedpandaConncetSecret = 'redpanda-connect-secret',
+}
+
+/**
+ * The Redpanda Connect Secret Manager introduces a new tab in Redpanda Connect.
+ * this logic determines which tab should be opened based on the `defaultTab`
+ * query parameter in the URL.
+ */
+const getDefaultView = (defaultView: string): { initialTab: ConnectView, redpandaConnectTab: ConnectView } => {
+
+    const showPipelines = Features.pipelinesApi
+    const showKafkaTab = {initialTab: ConnectView.kafkaConnect, redpandaConnectTab: ConnectView.RedpandaConnect}
+    const showRedpandaConnectTab = {initialTab: ConnectView.RedpandaConnect, redpandaConnectTab: ConnectView.RedpandaConnect}
+    if (!showPipelines) {
+        return showKafkaTab;
+    }
+
+    switch (defaultView) {
+        case 'kafka-connect':
+            return showKafkaTab;
+        case 'redpanda-connect':
+            return showRedpandaConnectTab;
+        case 'redpanda-connect-secret':
+            return {initialTab: ConnectView.RedpandaConnect, redpandaConnectTab: ConnectView.RedpandaConncetSecret};
+        default:
+            return showRedpandaConnectTab;
+    }
+}
+
+const WrapUseSearchParamsHook: FunctionComponent<{matchedPath: string}> = (props) => {
+    const {search}= useLocation();
+    const searchParams = new URLSearchParams(search);
+    const defaultTab = searchParams.get('defaultTab') || '';
+    return <KafkaConnectOverview defaultView={defaultTab} {...props}/>
```

Comment:
What happens if `defaultTab` is `''`? I think we need to return default to `ConnectView.RedpandaConnect` tab by default?

Reply:
it should default from `getDefaultView()`